granian/cli.py

```diff
@@ -1,135 +1,254 @@
-    ),
-    ssl_keyfile: Optional[Path] = typer.Option(
-        None, help='SSL key file', exists=True, file_okay=True, dir_okay=False, readable=True
-    ),
-    ssl_certificate: Optional[Path] = typer.Option(
-        None, help='SSL certificate file', exists=True, file_okay=True, dir_okay=False, readable=True
-    ),
-    url_path_prefix: Optional[str] = typer.Option(None, help='URL path prefix the app is mounted on'),
-    respawn_failed_workers: bool = typer.Option(
-        False,
-        '--respawn-failed-workers/--no-respawn-failed-workers',
-        help='Enable workers respawn on unexpected exit',
-        show_default='disabled',
-    ),
-    reload: bool = typer.Option(
-        False,
-        '--reload/--no-reload',
-        help="Enable auto reload on application's files changes (requires granian[reload] extra)",
-        show_default='disabled',
-    ),
-    process_name: Optional[str] = typer.Option(
-        None,
-        help='Set a custom name for processes (requires granian[pname] extra)',
-    ),
-    _: Optional[bool] = typer.Option(
-        None,
-        '--version',
-        callback=version_callback,
-        is_eager=True,
-        help='Shows the version and exit',
-        allow_from_autoenv=False,
-    ),
-):
+_AnyCallable = Callable[..., Any]
+FC = TypeVar('FC', bound=Union[_AnyCallable, click.Command])
+
+
+class EnumType(click.Choice):
+    def __init__(self, enum: Enum, case_sensitive=False):
+        self.__enum = enum
+        super().__init__(choices=[item.value for item in enum], case_sensitive=case_sensitive)
+
+    def convert(self, value, param, ctx):
+        if value is None or isinstance(value, Enum):
+            return value
+
+        converted_str = super().convert(value, param, ctx)
+        return self.__enum(converted_str)
+
+
+def pretty_print_default(value: Optional[bool]) -> Optional[str]:
+    """Print nicer message strings for default value."""
+    if isinstance(value, bool):
+        return 'enabled' if value else 'disabled'
+    if isinstance(value, Enum):
+        return value.value
+    return value
+
+
+def show_env_var(*param_decls: str, cls: Optional[Type[click.Option]] = None, **attrs: Any) -> Callable[[FC], FC]:
+    """This is the option generator from Click.
+
+    It has a very simple update so that the environment variable will also get added to the help text.
+    """
+    if cls is None:
+        cls = click.Option
+
+    def decorator(f: FC) -> FC:
+        # start change
+        env_name = param_decls[-1].lstrip('-').split('/')[0].replace('-', '_').upper()
+        attrs['help'] += f' [env var: GRANIAN_{env_name}]'
+        if 'default' in attrs:
+            attrs['show_default'] = pretty_print_default(attrs['default'])
+        # end change
+        click.decorators._param_memo(f, cls(param_decls, **attrs))
+        return f
+
+    return decorator
+
+
+# monkey patch so that the behaviour is more similar to the docs.
+click.option = show_env_var
```

Comment:
I just thought monkey patching `click.option` might not be a good idea, as several frameworks/apps used with Granian might depend on it.

Maybe we should just update `show_env_var` to be a `cli_option` we use in place of `click.option` that will just return `click.option` decorator patching the arguments in advance.

Reply:
That's true. Updated the PR.